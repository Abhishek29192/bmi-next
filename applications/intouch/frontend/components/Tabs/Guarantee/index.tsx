import React, { useState } from "react";
import Button from "@bmi/button";
import Modal from "@bmi/dialog";
import { useTranslation } from "next-i18next";
import { GetProjectQuery } from "../../../graphql/generated/operations";
import { checkProjectGuaranteeReview } from "../../../lib/utils/project";
import styles from "./styles.module.scss";
import { ApplyGuaranteeDialog } from "./ApplyGuaranteeDialog";
import { ProjectGuarantee } from "./ProjectGuarantee";
import SolutionGuaranteeReviewDialog from "./SolutionGuaranteeReviewDialog";

export type GuaranteeTabProps = {
  project: GetProjectQuery["project"];
};

export const GuaranteeTab = ({ project }: GuaranteeTabProps) => {
  const { t } = useTranslation("project-page");
  const [isDialogOpen, setDialogOpen] = useState(false);
  const [isModalOpen, setModalOpen] = useState(false);
  const [isReviewOpen, setReviewOpen] = useState(false);

  const { guarantees } = project;

  return (
    <div className={styles.main}>
      <div className={styles.header}>
        <Button onClick={() => setDialogOpen(true)}>
          {t("guarantee_tab.header")}
        </Button>
      </div>
      <div className={styles.body}>
        <ProjectGuarantee
          guarantees={guarantees.nodes}
          onReviewClick={() => setReviewOpen(true)}
          canGuaranteeBeSubmitted={checkProjectGuaranteeReview(project)}
        />
      </div>
      <ApplyGuaranteeDialog
        isOpen={isDialogOpen}
        project={project}
        onCloseClick={() => setDialogOpen(false)}
        onCompletedClick={() => {
          setDialogOpen(false);
          setModalOpen(true);
        }}
      />
      <SolutionGuaranteeReviewDialog
        isOpen={isReviewOpen}
        project={project}
        onCloseClick={() => setReviewOpen(false)}
      />
      <Modal open={isModalOpen} onCloseClick={() => setModalOpen(false)}>
        <Modal.Title hasUnderline>
          {t("guarantee_tab.dialog.title")}
        </Modal.Title>
        <Modal.Content>{t("guarantee_tab.dialog.description")}</Modal.Content>
        <Modal.Actions
          confirmLabel={t("guarantee_tab.dialog.confirm_label")}
          onConfirmClick={() => setModalOpen(false)}
        />
      </Modal>
    </div>
  );
};
