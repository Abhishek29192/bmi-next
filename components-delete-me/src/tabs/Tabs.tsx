import type { GridProps } from "@material-ui/core";
import {
  AppBarProps,
  Box,
  Tab as MaterialTab,
  TabProps as MaterialTabProps,
  Tabs as MaterialTabs,
  TabsProps as MaterialTabsProps
} from "@material-ui/core";
import classnames from "classnames";
import React from "react";
import Container, { Props as ContainerProps } from "../container/Container";
import Grid from "../grid/Grid";
import { transformHyphens } from "../utils/hyphenUtils";
import { useGlobalTabStyles } from "./styles";

type TabPanelProps = GridProps & {
  className?: string;
  children?: React.ReactNode;
  heading: string;
  index: any;
  value?: any;
};

const TabPanel = ({
  children,
  className,
  value,
  index,
  ...other
}: TabPanelProps) => {
  const globalClasses = useGlobalTabStyles();
  const formattedIndex =
    typeof index === "string" ? index.replace(/ /g, "-") : index;
  const formattedvalue =
    typeof value === "string" ? value.replace(/ /g, "-") : value;
  return (
    <Grid
      item
      className={classnames(globalClasses.TabPanel, className)}
      aria-labelledby={`tab-${formattedIndex}`}
      hidden={formattedvalue !== formattedIndex}
      id={`tabpanel-${formattedIndex}`}
      key={`tabpanel-${formattedIndex}`}
      role="tabpanel"
      data-testid={`tabpanel-${formattedIndex}`}
      {...other}
    >
      <Box className={globalClasses.TabPanelBox} p={3}>
        {children}
      </Box>
    </Grid>
  );
};

type TabsProps = MaterialTabsProps &
  Pick<AppBarProps, "color"> & {
    children:
      | (React.ReactElement<TabPanelProps> | React.ReactNode)
      | (React.ReactElement<TabPanelProps> | React.ReactNode)[];
    initialValue?: any;
    tabComponent?: React.ComponentType<any>; // TODO
    visibleUntil?: "sm" | "md";
    theme?: "primary" | "secondary";
    // NOTE: This is necessary until we upgrade to @material-ui/core@^5.0.0
    // see: https://github.com/mui-org/material-ui/issues/22452#issuecomment-685756045
    component?: React.ElementType;
    onChange?: (index: string) => void;
    maxWidth?: ContainerProps["maxWidth"];
  };

export type TabProps = { index?: number } & MaterialTabProps;

export const Tab = ({ index, label, ...rest }: TabProps) => {
  const globalClasses = useGlobalTabStyles();
  return (
    <MaterialTab
      aria-controls={`tabpanel-${index}`}
      id={`tab-${index}`}
      key={index}
      label={label}
      value={index}
      className={globalClasses.TabRoot}
      {...rest}
    />
  );
};

const Tabs = ({
  children,
  className,
  initialValue = 0,
  tabComponent: Tab = MaterialTab,
  visibleUntil,
  theme = "primary",
  onChange,
  maxWidth,
  ...other
}: TabsProps) => {
  const globalClasses = useGlobalTabStyles();
  const [value, setValue] = React.useState(initialValue);

  const handleChange = (
    _event: React.ChangeEvent<Record<string, unknown>>,
    newValue: any
  ) => {
    onChange && onChange(newValue);
    setValue(newValue);
  };

  const tabs = React.Children.toArray(children).filter(
    Boolean
  ) as React.ReactElement[];

  return (
    <div
      className={classnames(
        globalClasses.root,
        theme === "secondary" && globalClasses.secondary,
        visibleUntil && globalClasses[`visibleUntil${visibleUntil}`],
        className
      )}
    >
      <div className={globalClasses.TabsBar}>
        <Container maxWidth={maxWidth} className={globalClasses.ContainerRoot}>
          <MaterialTabs
            aria-label="tabs"
            indicatorColor="primary"
            onChange={handleChange}
            scrollButtons="on"
            TabScrollButtonProps={{ className: "tabsScrollBtn" }}
            textColor="primary"
            variant="scrollable"
            value={value}
            className={globalClasses.TabsRoot}
            {...other}
          >
            {tabs.map(({ props: { heading, index } }) => {
              const formattedIndex =
                typeof index === "string" ? index.replace(/ /g, "-") : index;
              return (
                <Tab
                  aria-controls={`tabpanel-${formattedIndex}`}
                  id={`tab-${formattedIndex}`}
                  key={formattedIndex}
                  label={transformHyphens(heading)}
                  value={index}
                  className={globalClasses.TabRoot}
                  data-testid={`tab-${formattedIndex}`}
                />
              );
            })}
          </MaterialTabs>
        </Container>
      </div>
      {tabs.map((child) => {
        const { index, ...other } = child.props;
        return React.cloneElement(child, {
          index,
          key: index,
          value,
          ...other
        });
      })}
    </div>
  );
};

Tabs.TabPanel = TabPanel;

export default Tabs;
