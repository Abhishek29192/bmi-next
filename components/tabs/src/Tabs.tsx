import {
  AppBarProps,
  Box,
  Tab as DefaultTab,
  Tabs as MaterialTabs,
  TabsProps as MaterialTabsProps
} from "@material-ui/core";
import Container, { Props as ContainerProps } from "@bmi/container";
import Grid, { GridProps } from "@bmi/grid";
import classnames from "classnames";
import React from "react";
import styles from "./Tabs.module.scss";

type TabPanelProps = GridProps & {
  className?: string;
  children?: React.ReactNode;
  heading: String;
  index: any;
  value?: any;
};

const TabPanel = ({
  children,
  className,
  value,
  index,
  ...other
}: TabPanelProps) => (
  <Grid
    item
    className={classnames(styles["TabPanel"], className)}
    aria-labelledby={`tab-${index}`}
    hidden={value !== index}
    id={`tabpanel-${index}`}
    role="tabpanel"
    {...other}
  >
    <Box className={styles["TabPanelBox"]} p={3}>
      {children}
    </Box>
  </Grid>
);

type TabsProps = MaterialTabsProps &
  Pick<AppBarProps, "color"> & {
    children:
      | (React.ReactElement<TabPanelProps> | React.ReactNode)
      | (React.ReactElement<TabPanelProps> | React.ReactNode)[];
    initialValue?: any;
    tabComponent?: React.ComponentType<any>; // TODO
    visibleUntil?: "sm" | "md";
    theme?: "primary" | "secondary";
    // NOTE: This is necessary until we upgrade to @material-ui/core@^5.0.0
    // see: https://github.com/mui-org/material-ui/issues/22452#issuecomment-685756045
    component?: React.ElementType;
    onChange?: (index: string) => void;
    maxWidth?: ContainerProps["maxWidth"];
  };

const Tabs = ({
  children,
  className,
  initialValue = 0,
  tabComponent: Tab = DefaultTab,
  visibleUntil,
  theme = "primary",
  onChange,
  maxWidth,
  ...other
}: TabsProps) => {
  const [value, setValue] = React.useState(initialValue);

  const handleChange = (_event: React.ChangeEvent<{}>, newValue: any) => {
    onChange && onChange(newValue);
    setValue(newValue);
  };

  const tabs = React.Children.toArray(children).filter(
    Boolean
  ) as React.ReactElement[];

  return (
    <div
      className={classnames(
        styles["Tabs"],
        styles[`Tabs--${theme}`],
        visibleUntil && styles[`Tabs--visible-until-${visibleUntil}`],
        className
      )}
    >
      <div className={styles["TabsBar"]}>
        <Container maxWidth={maxWidth}>
          <MaterialTabs
            aria-label="tabs"
            indicatorColor="primary"
            onChange={handleChange}
            scrollButtons="auto"
            textColor="primary"
            variant="scrollable"
            value={value}
            {...other}
          >
            {tabs.map(({ props: { heading, index } }) => (
              <Tab
                aria-controls={`tabpanel-${index}`}
                id={`tab-${index}`}
                key={index}
                label={heading}
                value={index}
              />
            ))}
          </MaterialTabs>
        </Container>
      </div>
      {tabs.map((child) => {
        const { index, ...other } = child.props;
        return React.cloneElement(child, {
          index,
          key: index,
          value,
          ...other
        });
      })}
    </div>
  );
};

Tabs.TabPanel = TabPanel;

export default Tabs;
