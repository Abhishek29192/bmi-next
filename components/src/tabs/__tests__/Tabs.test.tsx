import { fireEvent, render } from "@testing-library/react";
import React from "react";
import Tabs from "../Tabs";

describe("Tabs component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Tabs>
        <Tabs.TabPanel heading="Tab 1" index={0}>
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel heading="Tab 2" index={1}>
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });
  it("replaces spaces in tab and tab panel id attributes with hyphen", () => {
    const { getByTestId } = render(
      <Tabs>
        <Tabs.TabPanel
          heading="Tab 1"
          index={"some index 1"}
          key={"some key 1"}
        >
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel
          heading="Tab 2"
          index={"some index 2"}
          key={"some key 1"}
        >
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );
    const tabPanel1 = getByTestId("tabpanel-some-index-1");
    expect(tabPanel1.id).toEqual("tabpanel-some-index-1");
    const tab1 = getByTestId("tab-some-index-1");
    expect(tab1.id).toEqual("tab-some-index-1");
    const tabPanel2 = getByTestId("tabpanel-some-index-2");
    expect(tabPanel2.id).toEqual("tabpanel-some-index-2");
    const tab2 = getByTestId("tab-some-index-2");
    expect(tab2.id).toEqual("tab-some-index-2");
  });
  it("renders secondary theme correctly", () => {
    const { container } = render(
      <Tabs theme="secondary">
        <Tabs.TabPanel heading="Tab 1" index={0}>
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel heading="Tab 2" index={1}>
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });
  it("triggers onChange", () => {
    const onChange = jest.fn();
    const { getByText } = render(
      <Tabs initialValue="tabOne" theme="secondary" onChange={onChange}>
        <Tabs.TabPanel heading="Tab 1" index={"tabOne"}>
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel heading="Tab 2" index={"tabTwo"}>
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );
    const firstTabHeading = getByText("Tab 2");
    fireEvent.click(firstTabHeading);
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange.mock.calls).toMatchSnapshot();
  });
  it("renders correctly when one tab is false", () => {
    const { container } = render(
      <Tabs>
        <Tabs.TabPanel heading="Tab 1" index={0}>
          Content One
        </Tabs.TabPanel>
        {false}
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders correctly when visibleUntil is set to md ", () => {
    const { container } = render(
      <Tabs visibleUntil={"md"}>
        <Tabs.TabPanel heading="Tab 1" index={0}>
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel heading="Tab 2" index={1}>
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders correctly when visibleUntil is set to sm ", () => {
    const { container } = render(
      <Tabs visibleUntil={"sm"}>
        <Tabs.TabPanel heading="Tab 1" index={0}>
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel heading="Tab 2" index={1}>
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });
  it("should add hidden attribute for all tabs exepct active one and toggle hidden attribute on tabs active state change", () => {
    const { getByTestId } = render(
      <Tabs>
        <Tabs.TabPanel
          heading="Tab 1"
          index={"some index 1"}
          key={"some key 1"}
        >
          Content One
        </Tabs.TabPanel>
        <Tabs.TabPanel
          heading="Tab 2"
          index={"some index 2"}
          key={"some key 1"}
        >
          Content Two
        </Tabs.TabPanel>
      </Tabs>
    );

    const tab1 = getByTestId("tab-some-index-1");
    const tab2 = getByTestId("tab-some-index-2");
    const tabPanel1 = getByTestId("tabpanel-some-index-1");
    const tabPanel2 = getByTestId("tabpanel-some-index-2");

    fireEvent.click(tab1);

    expect(tabPanel1).toHaveProperty("hidden", false);
    expect(tabPanel2).toHaveProperty("hidden", true);

    fireEvent.click(tab2);

    expect(tabPanel1).toHaveProperty("hidden", true);
    expect(tabPanel2).toHaveProperty("hidden", false);
  });
});
