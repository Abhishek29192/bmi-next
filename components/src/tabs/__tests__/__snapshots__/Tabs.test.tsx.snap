// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders correctly 1`] = `
<div>
  <div
    class="Tabs Tabs--primary"
  >
    <div
      class="TabsBar"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root tabStyles-ContainerRoot-1 MuiContainer-maxWidthXl"
        >
          <div
            class="MuiTabs-root tabStyles-TabsRoot-2"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiTabs-scrollable"
              style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
            />
            <div
              class="MuiTabs-scroller MuiTabs-scrollable"
              style="margin-bottom: 0px;"
            >
              <div
                aria-label="tabs"
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-3 Mui-selected"
                  data-testid="tab-0"
                  id="tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 1
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="tabpanel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-3"
                  data-testid="tab-1"
                  id="tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 2
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-4 PrivateTabIndicator-colorPrimary-5 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="tab-0"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-0"
      heading="Tab 1"
      id="tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-8 TabPanelBox"
      >
        Content One
      </div>
    </div>
    <div
      aria-labelledby="tab-1"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-1"
      heading="Tab 2"
      hidden=""
      id="tabpanel-1"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-9 TabPanelBox"
      >
        Content Two
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs component renders correctly when one tab is false 1`] = `
<div>
  <div
    class="Tabs Tabs--primary"
  >
    <div
      class="TabsBar"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root tabStyles-ContainerRoot-37 MuiContainer-maxWidthXl"
        >
          <div
            class="MuiTabs-root tabStyles-TabsRoot-38"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiTabs-scrollable"
              style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
            />
            <div
              class="MuiTabs-scroller MuiTabs-scrollable"
              style="margin-bottom: 0px;"
            >
              <div
                aria-label="tabs"
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-39 Mui-selected"
                  data-testid="tab-0"
                  id="tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 1
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-40 PrivateTabIndicator-colorPrimary-41 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="tab-0"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-0"
      heading="Tab 1"
      id="tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-44 TabPanelBox"
      >
        Content One
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs component renders correctly when visibleUntil is set to md  1`] = `
<div>
  <div
    class="Tabs Tabs--primary Tabs--visible-until-md"
  >
    <div
      class="TabsBar"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root tabStyles-ContainerRoot-45 MuiContainer-maxWidthXl"
        >
          <div
            class="MuiTabs-root tabStyles-TabsRoot-46"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiTabs-scrollable"
              style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
            />
            <div
              class="MuiTabs-scroller MuiTabs-scrollable"
              style="margin-bottom: 0px;"
            >
              <div
                aria-label="tabs"
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-47 Mui-selected"
                  data-testid="tab-0"
                  id="tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 1
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="tabpanel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-47"
                  data-testid="tab-1"
                  id="tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 2
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-48 PrivateTabIndicator-colorPrimary-49 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="tab-0"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-0"
      heading="Tab 1"
      id="tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-52 TabPanelBox"
      >
        Content One
      </div>
    </div>
    <div
      aria-labelledby="tab-1"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-1"
      heading="Tab 2"
      hidden=""
      id="tabpanel-1"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-53 TabPanelBox"
      >
        Content Two
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs component renders correctly when visibleUntil is set to sm  1`] = `
<div>
  <div
    class="Tabs Tabs--primary Tabs--visible-until-sm"
  >
    <div
      class="TabsBar"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root tabStyles-ContainerRoot-54 MuiContainer-maxWidthXl"
        >
          <div
            class="MuiTabs-root tabStyles-TabsRoot-55"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiTabs-scrollable"
              style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
            />
            <div
              class="MuiTabs-scroller MuiTabs-scrollable"
              style="margin-bottom: 0px;"
            >
              <div
                aria-label="tabs"
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-56 Mui-selected"
                  data-testid="tab-0"
                  id="tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 1
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="tabpanel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-56"
                  data-testid="tab-1"
                  id="tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 2
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-57 PrivateTabIndicator-colorPrimary-58 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="tab-0"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-0"
      heading="Tab 1"
      id="tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-61 TabPanelBox"
      >
        Content One
      </div>
    </div>
    <div
      aria-labelledby="tab-1"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-1"
      heading="Tab 2"
      hidden=""
      id="tabpanel-1"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-62 TabPanelBox"
      >
        Content Two
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs component renders secondary theme correctly 1`] = `
<div>
  <div
    class="Tabs Tabs--secondary"
  >
    <div
      class="TabsBar"
    >
      <div
        class="Container"
      >
        <div
          class="MuiContainer-root tabStyles-ContainerRoot-19 MuiContainer-maxWidthXl"
        >
          <div
            class="MuiTabs-root tabStyles-TabsRoot-20"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiTabs-scrollable"
              style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
            />
            <div
              class="MuiTabs-scroller MuiTabs-scrollable"
              style="margin-bottom: 0px;"
            >
              <div
                aria-label="tabs"
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-21 Mui-selected"
                  data-testid="tab-0"
                  id="tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 1
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="tabpanel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tabStyles-TabRoot-21"
                  data-testid="tab-1"
                  id="tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab 2
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-22 PrivateTabIndicator-colorPrimary-23 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiTabScrollButton-root tabsScrollBtn Mui-disabled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="tab-0"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-0"
      heading="Tab 1"
      id="tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-26 TabPanelBox"
      >
        Content One
      </div>
    </div>
    <div
      aria-labelledby="tab-1"
      class="MuiGrid-root TabPanel MuiGrid-item"
      data-testid="tabpanel-1"
      heading="Tab 2"
      hidden=""
      id="tabpanel-1"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-27 TabPanelBox"
      >
        Content Two
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs component triggers onChange 1`] = `
Array [
  Array [
    "tabTwo",
  ],
]
`;
